File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.py
def solution(n):
    arr = [0]*2000
    arr[0] = 1
    arr[1] = 2
    
    for i in range(2, n):
        arr[i] = (arr[i-1] + arr[i-2])%1234567
    
    return arr[n-1]

File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.java
class Solution {
    public long solution(int n) {
        int[] dp = new int[2000];
        
        dp[0] = 1;
        dp[1] = 2;
        
        for (int i=2; i<n; i++) {
            dp[i] = (dp[i-1] + dp[i-2]) % 1234567;
        }
        
        return dp[n-1];
    }
}